### PR TITLE
tensorflow: fix build

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

With gcc 10, tensorflow 2 broke. Also remove tensorflow 1 which does no longer work with our current default python3 version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -7352,62 +7352,39 @@ in {
...
 
-  tensorflow-estimator_2 = callPackage ../development/python-modules/tensorflow-estimator/2 { };
+  tensorflow-estimator_2 = callPackage ../development/python-modules/tensorflow-estimator { };
 
   tensorflow-estimator = self.tensorflow-estimator_1;
 
   tensorflow-probability = callPackage ../development/python-modules/tensorflow-probability { };
 
-  tensorflow = self.tensorflow_1;
-  tensorflow_1 = self.tensorflow-build_1;
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:7403]: Please add an alias with a throw for tensorflow_1.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -7352,62 +7352,39 @@ in {
...
+    cudnn = pkgs.cudnn_cudatoolkit_11_0;
     nccl = pkgs.nccl_cudatoolkit_11;
     openssl = pkgs.openssl_1_1;
     inherit (pkgs.darwin.apple_sdk.frameworks) Foundation Security;
     inherit (pkgs) flatbuffers;
   };
 
-  tensorflow-build = self.tensorflow-build_1;
-
-  tensorflow-estimator_1 = callPackage ../development/python-modules/tensorflow-estimator/1 { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:7394]: Please add an alias with a throw for tensorflow-estimator_1.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -7352,62 +7352,39 @@ in {
...
-  tensorflow-bin_2 = callPackage ../development/python-modules/tensorflow/2/bin.nix {
-    cudaSupport = pkgs.config.cudaSupport or false;
-    inherit (pkgs.linuxPackages) nvidia_x11;
-    cudatoolkit = pkgs.cudatoolkit_10;
-    cudnn = pkgs.cudnn_cudatoolkit_10;
-  };
-
-  tensorflow-bin = self.tensorflow-bin_1;
-
-  tensorflow-build_1 = callPackage ../development/python-modules/tensorflow/1 {
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:7371]: Please add an alias with a throw for tensorflow-build_1.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -7352,62 +7352,39 @@ in {
 
   tensorboardx = callPackage ../development/python-modules/tensorboardx { };
 
-  tensorflow-bin_1 = callPackage ../development/python-modules/tensorflow/1/bin.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:7355]: Please add an alias with a throw for tensorflow-bin_1.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/top-level/python-packages.nix:7403]: I don't think we do this in general for version suffixes for example the linux kernel or compiler versions.

**Pull request merged**